File: sort/InsertionSort.py
data = [3, 45, 45, 2, 6, 8, 3, 7, 9, 6, 4, 7, 8, 6, 87, 0, 6, 6]

def insertion_sort(data):
    for i in range(1, len(data)):           #for 1 to length (unsorted data)
        temp = data[i]                          # key = element i
        j = i - 1                               # iterate without repetition
        while j >= 0 and temp < data[j]:        # leftside loop (sorted data) (stops)
            data[j + 1] = data[j]               # nudge to left
            j -= 1                              # point to open space
        data[j + 1] = temp                      # set new temp
    return data

print(insertion_sort(data))

File: sort/InsertionSort.java
package sort;

public class InsertionSort{
    public static void main(String[] args) {
        int[] data = {15,5226,73,48,58,59,96,6,9584,73,748,737,47,3};

        data = insertion_sort(data);

        for(int i : data) System.out.print( i + "   ");
    }

    public static int[] insertion_sort(int[] data){
        for(int i = 1; i < data.length; i++){
            int temp = data[i];
            int j = i - 1;
            while(j >= 0 && temp < data[j]){
                data[j + 1] = data[j];
                j--;
            }
            data[j + 1] = temp;
        }
        return data;
    }
}